src/Latte/Parser/TemplateTypeParser.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rector\Nette\Latte\Parser;
+
+use Nette\Utils\Strings;
+use PHPStan\BetterReflection\Reflection\ReflectionClass;
+use PHPStan\BetterReflection\Reflection\ReflectionNamedType;
+use Rector\Nette\ValueObject\LatteVariableType;
+
+final class TemplateTypeParser
+{
+    /**
+     * @return LatteVariableType[]
+     */
+    public function parse(string $content): array
+    {
+        $templateTypePattern = '#{templateType (?P<template>.*?)}#';
+        $templateTypeMatch = Strings::match($content, $templateTypePattern);
+        if (! isset($templateTypeMatch['template'])) {
+            return [];
+        }
+
+        $variableTypes = [];
+        $reflectionClass = ReflectionClass::createFromName($templateTypeMatch['template']);
```

Comment:
What if the class is not found? 

We use static reflection in Rector, so this would not work if class is not autoloaded.

Instead use `ReflectionProvider` from PHPStan, something like this:

```php
$templateClassName = $templateTypeMatch['template'];
if (! $this->reflectionProvider->hasClass($templateClassName)) {
   return [];
}

$classReflection = $this->refelectionProvider->getClass($templateClassName);
```

Reply:
Hm, you're right, but this $classReflection (PHPStan\Reflection\ClassReflection) hasn't method getProperties() or something similar for load all properties in class.

Reply:
For now I handled it with try catch - unit test added. If reflectionProvider can be used, please show me some example how I can get all properties of reflected class and I'll rewrite the code.

Reply:
I see, let's give it a try then.